src/utils/process.rs

```diff
@@ -23,23 +25,48 @@ pub struct CommandLine {
     last_arg: Option<String>,
 }
 
-pub fn calling_process() -> Option<Cow<'static, CallingProcess>> {
-    #[cfg(not(test))]
-    {
-        CACHED_CALLING_PROCESS.as_ref().map(Cow::Borrowed)
-    }
-    #[cfg(test)]
-    {
-        determine_calling_process().map(Cow::Owned)
-    }
+lazy_static! {
+    static ref CALLER: Arc<(Mutex<CallingProcess>, Condvar)> =
+        Arc::new((Mutex::new(CallingProcess::Pending), Condvar::new()));
 }
 
-lazy_static! {
-    static ref CACHED_CALLING_PROCESS: Option<CallingProcess> = determine_calling_process();
+pub fn start_determining_calling_process_in_thread() {
+    // Do not keep the join handle so the main thread can exit early if
+    // it does not need to know its parent process.
```

Comment:
This comment is referring to the fact that there is no return value of this function (and that we use the `spawn` method on the builder as opposed to the `thread::spawn` function), right?

Reply:
The builder is used to set the thread name - in case someone without access to the sources wonders why delta looks at all the processes. It could also return a `JoinHandle`, but this is ignored. I updated the comments to reflect this.